### ISSUE TITLE
weight decay feature

### ISSUE THREAD

**Issue by WenchenLi** [aigamedev/scikit-neuralnetwork#44]
Hi,
Is it possible to implement weight decay to control the over-fitting problem although these is dropout.

**Comment by alexjc** [comment]
How would you see this working, as a decay per epoch?  Would it be a relative decay or absolute?

**Comment by ssamot** [comment]
I am pretty sure pylearn2 has weight decay as a feature - it works layer-wise, similiar to dropout. I think this is very important feature because the only way you can regularise now is dropout - and l2 would be nice to have.

**Comment by alexjc** [comment]
Maybe an additional `decay` parameter for the Layer objects? Are per-layer learning rate decay common at all? Never used them.
